vms/platformvm/vm_regression_test.go

```diff
@@ -2130,12 +2131,18 @@ func checkValidatorBlsKeyIsSet(
 	}
 
 	val, found := vals[nodeID]
-	if !found {
+	switch {
+	case !found:
 		return database.ErrNotFound
-	}
-	if val.PublicKey != expectedBlsKey {
```

Comment:
This check assumes that the `PublicKey` pointer is equal, not the actual public key... Which means this is requiring the key to have been cached and not re-parsed.